logstash-core/spec/logstash/util/wrapped_synchronous_queue_spec.rb

```diff
@@ -1,16 +1,15 @@
 # encoding: utf-8
 require "spec_helper"
-require "logstash/util/wrapped_synchronous_queue"
 require "logstash/instrument/collector"
 
-describe LogStash::Util::WrappedSynchronousQueue do
+describe LogStash::WrappedSynchronousQueue do
 
-  subject {LogStash::Util::WrappedSynchronousQueue.new(5)}
+  subject {LogStash::WrappedSynchronousQueue.new(5)}
 
   describe "queue clients" do
     context "when requesting a write client" do
       it "returns a client" do
-        expect(subject.write_client).to be_a(LogStash::Util::WrappedSynchronousQueue::WriteClient)
+        expect(subject.write_client).to be_a(LogStash::MemoryWriteClient)
```

Comment:
when we change both specs and implementation in a single commit, it becomes hard to reason about what is changing (and to prove what is remaining the same).

Would it be possible to pull the java implementation into a matching namespace, so that we could prove we've got a 1:1 replacement?

Reply:
I'm not sure I follow? The behavioral tests in the spec are all untouched and the new JRuby extension class passes the same tests as the original JRuby class.